### PR TITLE
Update NFS testing image to use ganesha, to work on GCI

### PR DESCRIPTION
**Pull request by wongma7**
Discussed here https://github.com/kubernetes/kubernetes/pull/34983

@vishh this is the image that will work with the containerized mounter on GCI.

~~The tests will not pass with the 1st commit alone, I have to~~
- mount hostpath volumes (tmpfs) for the nfs-server pod to export out of as seen here https://github.com/wongma7/kubernetes/commit/388603619af53a71a6b75291273ae9b53d993d93#diff-d8068327d664ac7a6620dc0b42d31840R421. 
- and of course also have to change the test tags in volumes.go and persistent_volumes.go to `0.8` for when the image is pushed.

~~working on the above now.~~ edit:done 

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/35469)

### CODE REVIEW

**Comment by wongma7** [comment]
btw I don't think mountd is needed either for NFSv4 but didn't remember to remove the port stuff for it.

**Comment by vishh** [comment]
ok to test

File: test/images/volumes-tester/nfs/Dockerfile [hunk 1]
```diff
@@ -12,16 +12,33 @@
...
+
+# install nfs-ganesha
+RUN dnf install -y tar gcc cmake autoconf libtool bison flex make gcc-c++ krb5-devel dbus-devel && dnf clean all \
+    && curl -L https://github.com/nfs-ganesha/nfs-ganesha/archive/V2.4.0.3.tar.gz | tar zx \
+    && curl -L https://github.com/nfs-ganesha/ntirpc/archive/v1.4.1.tar.gz | tar zx \
+    && rm -r nfs-ganesha-2.4.0.3/src/libntirpc \
+    && mv ntirpc-1.4.1 nfs-ganesha-2.4.0.3/src/libntirpc \
+    && cd nfs-ganesha-2.4.0.3 \
+    && sed -i '20i#cmakedefine _NO_PORTMAPPER 1' src/include/config-h.in.cmake \
+    && cmake -DCMAKE_BUILD_TYPE=Release -DBUILD_CONFIG=vfs_only -D_NO_PORTMAPPER=ON src/ \
```
Comment (wongma7) [inline test/images/volumes-tester/nfs/Dockerfile]: I `dnf remove` the dependencies in the last line of this RUN step, so the resulting container won't have all those dependencies I pulled in just for compilation.

Another thing I could do to reduce the size is delete the sources after ganesha has been built, will work on that change now.

**Comment by wongma7** [comment]
OK, I now delete the sources. compressed size is 212 MB is that too big?

While I was there I did minor cleanup/maintain changes to get rid of the ugly `sed` line, by bumping versions. Will do another test that this actually works with the gci mounter just in case those changes broke something.

**Comment by vishh** [comment]
Sweet. Thanks! What is the current total size of the image? Is it `212 MB`?

**Comment by wongma7** [comment]
`docker images` says 387.2 MB. 

i got the 212 MB from docker hub, i guess that's the amount one would have to download.

**Comment by wongma7** [comment]
k, image still works with containerized mounter

**Comment by wongma7** [comment]
I don't think this is needed anymore, right? You have got NFS server container images working for all testing needs?

I still think it would be beneficial because it's more "container-y", runs a userspace NFS server instead of starting kernel stuff. But it's also less maintainable

**Comment by jingxu97** [comment]
close it for now.

**Pull request closed without merging**